src/Cache/_ScsDataClient.php

```diff
@@ -120,9 +132,90 @@ public function delete(string $cacheName, string $key) : CacheDeleteResponse
         } catch (SdkError $e) {
             return new CacheDeleteResponseError($e);
         } catch (\Exception $e){
-            return new CacheGetResponseError(new UnknownError($e->getMessage()));
+            return new CacheDeleteResponseError(new UnknownError($e->getMessage()));
         }
         return new CacheDeleteResponseSuccess();
     }
 
+    public function listFetch(string $cacheName, string $listName) : CacheListFetchResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateListName($listName);
+            $listFetchRequest = new _ListFetchRequest();
+            $listFetchRequest->setListName($listName);
+            $call = $this->grpcManager->client->ListFetch(
+                $listFetchRequest, ["cache"=>[$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]
+            );
+            $response = $this->processCall($call);
+        } catch (SdkError $e) {
+            return new CacheListFetchResponseError($e);
+        } catch (\Exception $e){
+            return new CacheListFetchResponseError(new UnknownError($e->getMessage()));
+        }
+        if (!$response->hasFound())
+        {
+            return new CacheListFetchResponseMiss();
+        }
+        return new CacheListFetchResponseHit($response);
+    }
+
+    public function listPushFront(
+        string $cacheName, string $listName, string $value, bool $refreshTtl, ?int $truncateBackToSize=null, ?int $ttlSeconds=null
+    ) : CacheListPushFrontResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateListName($listName);
+            $itemTtlSeconds = $ttlSeconds ? $ttlSeconds : $this->defaultTtlSeconds;
+            validateTtl($itemTtlSeconds);
+            $listPushFrontRequest = new _ListPushFrontRequest();
+            $listPushFrontRequest->setListName($listName);
+            $listPushFrontRequest->setValue($value);
+            $listPushFrontRequest->setRefreshTtl($refreshTtl);
+            $listPushFrontRequest->setTtlMilliseconds($itemTtlSeconds * 1000);
```

Comment:
I recommend pulling the seconds -> ms conversion to a separate function

Reply:
Good idea, that let me DRY out some other stuff too. Thanks for the suggestion. Fixed in 18c813c.